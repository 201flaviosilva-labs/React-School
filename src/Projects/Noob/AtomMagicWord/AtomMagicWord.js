import React from "react";
import { Provider, atom, useAtom } from "jotai";
import NoobProject from "../../../Pages/Noob/components/Project";

import "./style.css";

const textAtom = atom("hello");
const textLenAtom = atom((get) => get(textAtom).length);
const uppercaseAtom = atom((get) => get(textAtom).toUpperCase());

function Input() {
	const [text, setText] = useAtom(textAtom);
	return <input value={text} onChange={(e) => setText(e.target.value)} />;
};

function CharCount() {
	const [len] = useAtom(textLenAtom);
	return <div>Comprimento: {len}</div>;
};

function Uppercase() {
	const [uppercase] = useAtom(uppercaseAtom);
	return <div>Mai√∫sculas: {uppercase}</div>;
};

export default function AtomMagicWord() {
	return (
		<NoobProject title="Atom Magic Word" className="NoobAtomMagicWord">
			<Provider>
				<Input />
				<CharCount />
				<Uppercase />
			</Provider>
		</NoobProject>
	);
}
