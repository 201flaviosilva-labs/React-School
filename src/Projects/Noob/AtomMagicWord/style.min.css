.NoobAtomMagicWord input{width:100%;padding:2px 4px;border:1px solid black;border-radius:4px;text-align:center;margin-bottom:4px}
