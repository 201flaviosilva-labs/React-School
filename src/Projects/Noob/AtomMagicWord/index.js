import AtomMagicWord from "./AtomMagicWord";
export default AtomMagicWord;
