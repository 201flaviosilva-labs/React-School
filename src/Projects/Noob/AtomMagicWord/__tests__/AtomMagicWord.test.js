import React from "react";
import { shallow } from "enzyme";
import AtomMagicWordIndex from "../index";
import AtomMagicWord from "../AtomMagicWord";


describe("<AtomMagicWordIndex />", () => {
	test("Expect to render correctly", () => {
		const app = shallow(<AtomMagicWordIndex />);
		expect(app).toMatchSnapshot();
	});

	test("Expect to render correctly", () => {
		const app = shallow(<AtomMagicWord />);
		expect(app).toMatchSnapshot();
	});
});
