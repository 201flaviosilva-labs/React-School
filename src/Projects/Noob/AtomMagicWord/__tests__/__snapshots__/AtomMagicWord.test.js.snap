// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AtomMagicWordIndex /> Expect to render correctly 1`] = `ShallowWrapper {}`;

exports[`<AtomMagicWordIndex /> Expect to render correctly 2`] = `ShallowWrapper {}`;
